.github/workflows/ci.yml

```diff
@@ -62,10 +79,25 @@ jobs:
         run: composer install --prefer-dist --no-progress --optimize-autoloader --classmap-authoritative
         working-directory: tools/phar
 
-      - name: Compile phar Linux
+      - name: Build Castor phar for Linux
         run: bin/castor castor:phar:linux
         shell: bash
-        if: matrix.phar
+        if: matrix.castor.method == 'phar' || matrix.castor.method == 'binary'
+
+      - name: Restore PHP static building artifacts cache
+        uses: actions/cache/restore@v4
+        with:
+          path: |
+            /home/runner/.cache/castor/castor-php-static-compiler
+          # https://github.com/actions/cache/blob/main/tips-and-workarounds.md#update-a-cache
+          key: dynamic-key-${{ github.run_id }}
+          restore-keys: |
+            php-static-building-artifacts-cache-
```

Comment:
This was my initial idea, unfortunately it doesn't work as expected.

When the restore action with `key: X` is executed, it looks for a `cache[X]` and use it (hit) if it exists.
When the save action with `key: X` is executed, it tries to save  `cache[X]` but if fails if it already exists, **it doesn't update the cache key**.

And that's not what we want because when a new arg is added to the build command, a new directory is created: `/home/runner/.cache/castor/castor-php-static-compiler/{new_compile_command_build_hash}/*` and we want to update the cache with this new build artifacts.

If the current config what's happening is:

Run 1
- Restore Action:
   -   Does dynamic-key-${{ github.run_id }} cache hit ? No
   -   Does php-static-building-artifacts-cache-* cache hit ? No
- PHP Build
   -  Generate artifacts in `/home/runner/.cache/castor/castor-php-static-compiler/{compiler_test_command}/*`
   -  Generate artifacts in `/home/runner/.cache/castor/castor-php-static-compiler/{build_compiler_for_CASTOR_BIN}/*`
- Save Action:
   - Save `/home/runner/.cache/castor/castor-php-static-compiler/*` as `key: php-static-building-artifacts-cache-XXX` where XXX is hash(castor-php-static-compiler/*)

Run 2
- Restore Action:
   -   Does dynamic-key-${{ github.run_id }} cache hit ? No
   -   Does php-static-building-artifacts-cache-* cache hit ? Yes it founds php-static-building-artifacts-cache-XXX
- PHP Build
   -  Use artifacts cached in `/home/runner/.cache/castor/castor-php-static-compiler/**/*`, no build
compiler/{build_compiler_for_CASTOR_BIN}/*`
- Save Action:
   - Tries to save `/home/runner/.cache/castor/castor-php-static-compiler/*` as `key: php-static-building-artifacts-cache-XXX` but fails. It's ok no new artifacts were generated anyway.

Run 3, new compilation is added or options of existing build are changed in `CompileCommandTest` for example:
- Restore Action:
   -   Does dynamic-key-${{ github.run_id }} cache hit ? No
   -   Does php-static-building-artifacts-cache-* cache hit ? Yes, restore from Run 1
- PHP Build
   -  Build hash is different than Run 1, generate artifacts in `/home/runner/.cache/castor/castor-php-static-compiler/{compiler_test_command}/*`
   -   Build hash is same as Run 1, reuse artifacts in `/home/runner/.cache/castor/castor-php-static-compiler/{build_compiler_for_CASTOR_BIN}/*`
- Save Action:
   - Save `/home/runner/.cache/castor/castor-php-static-compiler/*` as `key: php-static-building-artifacts-cache-YYY` where YYY is a new hash of `castor-php-static-compiler/*` because new artifacts where generated since XXX

Reply:
There is two folders 

> * Generate artifacts in /home/runner/.cache/castor/castor-php-static-compiler/{compiler_test_command}/*
> * Generate artifacts in /home/runner/.cache/castor/castor-php-static-compiler/{build_compiler_for_CASTOR_BIN}/*

because 
* there is one for castor itself?
* there is another one for tests/CompileCommandTest.php

If yes, I almost got it, but not perfectly :) 
If not I'm totally lost 😅 


Anyway, I'll approve the PR, play a bit with it, with the cache, etc :) 

Thanks

Reply:
Yes exactly! Both build are not using the same extensions, the one in the test has less extension, it just makes sure it compiles and can run a hello world.

The compiled castor has more extensions because it needs to rune the whole suit of examples.